wrapper/src/test/java/integration/refactored/container/tests/HikariTests.java

```diff
@@ -110,4 +139,191 @@ public void testOpenConnectionWithDataSourceClassName() throws SQLException {
     assertTrue(conn.isValid(10));
     conn.close();
   }
+
+  /**
+   * After getting successful connections from the pool, the cluster becomes unavailable.
+   */
+  @TestTemplate
+  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  public void test_1_1_hikariCP_lost_connection() throws SQLException {
+    Properties customProps = new Properties();
+    customProps.setProperty(FailoverConnectionPlugin.FAILOVER_TIMEOUT_MS.name, "1");
+    PGProperty.SOCKET_TIMEOUT.set(customProps, "1");
+    createDataSource(customProps);
+    HikariDataSource ds = new HikariDataSource();
+    try (Connection conn = ds.getConnection()) {
+      assertTrue(conn.isValid(5));
+
+      ProxyHelper.disableAllConnectivity();
+
+      assertThrows(FailoverFailedSQLException.class, () -> queryInstanceId(conn));
+      assertFalse(conn.isValid(5));
+    }
+
+    assertThrows(SQLTransientConnectionException.class, () -> ds.getConnection());
+  }
+
+  /**
+   * After getting a successful connection from the pool, the connected instance becomes unavailable and the
+   * connection fails over to another instance. A connection is then retrieved to check that connections
+   * to failed instances are not returned.
+   */
+  @TestTemplate
+  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  public void test_1_2_hikariCP_get_dead_connection() throws SQLException {
+    ProxyHelper.disableAllConnectivity();
+
+    HikariDataSource ds = new HikariDataSource();
+
+    List<TestInstanceInfo> instances = TestEnvironment.getCurrent().getInfo().getDatabaseInfo().getInstances();
+
+    String writerIdentifier = instances.get(0).getEndpoint();
```

Comment:
The implication that the first endpoint is a writer should be embedded somewhere else.

Reply:
line 61 contains @ExtendWith(MakeSureFirstInstanceWriterExtension.class) so that the first endpoint is a writer